build.rs

```diff
@@ -1,16 +1,19 @@
+use std::env;
+
 fn main() {
-    if libcrux_platform::simd128_support() {
+    let target_arch = env::var("CARGO_CFG_TARGET_ARCH").unwrap();
+
+    if libcrux_platform::simd128_support() && target_arch == "x86_64" {
         println!("cargo:rustc-cfg=simd128");
     }
-    if libcrux_platform::simd256_support() {
+    if libcrux_platform::simd256_support() && target_arch == "x86_64" {
         println!("cargo:rustc-cfg=simd256");
     }
     if libcrux_platform::bmi2_adx_support() {
         println!("cargo:rustc-cfg=bmi2");
         println!("cargo:rustc-cfg=adx");
     }
-    #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
```

Comment:
I think we can do further cleanup by removing checks of `x86` arch along with features like `aesni` and `avx` in `sys/hacl/build.rs`